src/main/java/use_case/registration/RegistrationOutputBoundary.java

```diff
@@ -0,0 +1,9 @@
+package use_case.registration;
+
+public interface RegistrationOutputBoundary {
+    void prepareSuccessView(RegistrationOutputData outputData);
+
+    void prepareFailView(String errorMessage);
```

Comment:
Refer to my comment on the interactor. We should define a custom "prepare fail view" method for each type of failure (invalid username, duplicate username, etc.). Again refer to how buy stock use case is implemented and feel free to ask any questions.